FEM/fem_ele_vec.cpp

```diff
@@ -753,55 +731,37 @@ void CFiniteElementVec::ComputeStrain(const int ip)
  **************************************************************************/
 double CFiniteElementVec::CalDensity()
 {
-	double rho;
-	// OK_MFP
-	//--------------------------------------------------------------------
-	// MFP fluid properties
-	double density_fluid = 0.0;
-	double porosity = 0.0;
-	// double p_g = 0.0;
-	double Sw = 0.0;
-	int no_phases = (int)mfp_vector.size();
-	int i = 0, phase = 0;
-
-	rho = 0.0;
 	if (F_Flag)
 	{
-		if ((no_phases > 0) && (no_phases > phase))
-			density_fluid = m_mfp->Density();
-
-		// OK_MMP
 		//--------------------------------------------------------------------
-		// MMP medium properties
-		porosity = m_mmp->Porosity(this);
 		// Assume solid density is constant. (*smat->data_Density)(0)
 		if (smat->Density() > 0.0)
 		{
-			Sw = 1.0; // JT, should be 1.0, unless multiphase (calculate below) (if unsaturated, fluid density would be
-			          // negligible... so still works)
-			if (Flow_Type > 0 && Flow_Type != 10)
-			{
-				Sw = 0.; // WW
-				for (i = 0; i < nnodes; i++)
-					Sw += shapefct[i] * _nodal_S[i];
-			}
-			rho = (1. - porosity) * fabs(smat->Density()) + porosity * Sw * density_fluid;
+			const double density_fluid = (mfp_vector.empty()) ?  m_mfp->Density() : 0.0;
```

Comment:
i am puzzled here. why should the fluid density be zero if mfp_vector is not empty?

Reply:
Sorry, it is wrong. Removed the ternary operator.